core/src/main/java/org/infinispan/configuration/cache/PartitionHandlingConfigurationBuilder.java

```diff
@@ -27,8 +26,19 @@ public PartitionHandlingConfigurationBuilder(ClusteringConfigurationBuilder buil
    /**
     * @param enabled If {@code true}, partitions will enter degraded mode. If {@code false}, they will keep working independently.
     */
+   @Deprecated
    public PartitionHandlingConfigurationBuilder enabled(boolean enabled) {
-      attributes.attribute(ENABLED).set(enabled);
+      type(enabled ? PartitionHandling.DENY_ALL : PartitionHandling.ALLOW_ALL);
+      return this;
+   }
+
+   public PartitionHandlingConfigurationBuilder type(PartitionHandling partitionHandling) {
```

Comment:
@rvansa I agree that the naming can be improved. I will go with `whenSplit(DENY_READ_WRITES | ALLOW_READS | ALLOW_ALL)`